frontend/javascripts/oxalis/model/sagas/volumetracing_saga.js

```diff
@@ -919,11 +940,39 @@ export function* maintainHoveredSegmentId(): Saga<void> {
   yield _takeLatest("SET_MOUSE_POSITION", updateHoveredSegmentId);
 }
 
+function* maintainContourGeometry(): Saga<void> {
+  yield* take("SCENE_CONTROLLER_READY");
+  const SceneController = yield* call(getSceneController);
+  const { contour } = SceneController;
+
+  while (true) {
+    yield* take(["ADD_TO_LAYER", "RESET_CONTOUR"]);
+    const isTraceToolActive = yield* select(state => isTraceTool(state.uiInformation.activeTool));
+    const volumeTracing = yield* select(getActiveSegmentationTracing);
+    if (!volumeTracing) {
```

Comment:
You could add `|| !isTraceToolActive` here to avoid the large indented if block below.